controllers/ovndbcluster_controller.go

```diff
@@ -719,6 +719,10 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 	// cluster member so it can be resolved from outside cluster (edpm nodes)
 	if instance.Spec.NetworkAttachment != "" {
 		var dnsIPsList []string
+		if len(podList.Items) < int(*(instance.Spec.Replicas)) {
```

Comment:
This code will bail out without creating any DNSData records.

An argument can be made that it's better to create DNS records for pods that are already provisioned, even if not all of them are provisioned, yet. (so e.g. if there are 5 pods; but replicas is 7, it would be better to create DNSData records for the 5 pods and handle the rest later, on the next iteration.)

What do you think?

Reply:
Yep, I suppose that it better to have something that is slightly working than something that is not working at all. The important thing is to return error to ensure that the reconcile loop is triggered again.
The only "downside" is that the DNSData will be created every time (Can't do some check that if DNSData exist to skip it, since the fact that on every reconcile I recreate the DNSData handles the use case where pod gets restarted with a different IP)

Reply:
Adding comment regarding implementation, Need to cast all values from int to float as [Min](https://pkg.go.dev/math#Min) only accepts floats.
Also golang has a buildin [min](https://pkg.go.dev/builtin#min) function, but it was added in go1.21, currently on [go.mod](https://github.com/openstack-k8s-operators/ovn-operator/blob/main/go.mod#L3) it's defined that we use go1.20. Code not pretty but should do the work. I'm open if there's a better way.